src/corva/handlers.py

```diff
@@ -544,3 +562,55 @@ def _get_custom_event_type_by_raw_aws_event(
         if events:
             return event_type, handler
     return None, None
+
+
+def _merge_events(aws_event: Any, data_transformation_type: Type[RawBaseEvent]) -> Any:
+    """
+    Merges incoming aws_events into one.
+    Merge happens differently, depending on app type.
+    Only "scheduled" and "stream" type of apps can be processed here.
+    If somehow any other type is passed - raise an exception
+    """
+    if data_transformation_type == RawScheduledEvent:
+        # scheduled event
+        if not isinstance(aws_event[0], dict):
+            aws_event = list(itertools.chain(*aws_event))
+        is_depth = aws_event[0]["scheduler_type"] == SchedulerType.data_depth_milestone
+        event_start, event_end = (
+            ("top_depth", "bottom_depth")
+            if is_depth
+            else ("schedule_start", "schedule_end")
+        )
+        min_event_start, max_event_end = (
+            aws_event[0][event_start],
+            aws_event[0].get(event_end),
+        )
+        for event in aws_event[1:]:
+            if event[event_start] < min_event_start:
+                min_event_start = event[event_start]
+            if (
+                max_event_end
```

Comment:
If `aws_event[0].get(event_end)` returns `None`, `max_event_end` will never be reassigned